platform/android/src/test/benchmark_runner.cpp

```diff
@@ -1,10 +1,312 @@
+        std::string fileName = jstringToStdString(env, static_cast<jstring>(env->CallObjectMethod(f, fileGetName)));
+
+        if (isDir) {
+            if (!(env->CallBooleanMethod(f, fileIsDirectory))) {
+                jmethodID mkdirs = env->GetMethodID(fileClass, "mkdirs", "()Z");
+                bool success = (env->CallBooleanMethod(f, mkdirs));
+                std::string fileNameStr =
+                    jstringToStdString(env, static_cast<jstring>(env->CallObjectMethod(f, fileGetName)));
+
+                if (!success) {
+                    mbgl::Log::Warning(
+                        mbgl::Event::General, "Failed to create folder entry %s from zip", fileNameStr.c_str());
+                }
+            }
+        } else if (!(env->CallBooleanMethod(f, fileExists))) {
+            bool success = env->CallBooleanMethod(f, createNewFile);
+            std::string fileNameStr =
+                jstringToStdString(env, static_cast<jstring>(env->CallObjectMethod(f, fileGetName)));
+
+            if (!success) {
+                mbgl::Log::Warning(
+                    mbgl::Event::General, "Failed to create folder entry %s from zip", fileNameStr.c_str());
+                continue;
+            }
+
+            JavaWrapper<jobject> fout(env, env->NewObject(fileOutputStream, foutCtor, f));
+            JavaWrapper<jbyteArray> jBuff(env, env->NewByteArray(2048));
+
+            int count;
+            while ((count = env->CallIntMethod(zipIn.get(), zinRead, jBuff.get())) != -1) {
+                env->CallVoidMethod(fout.get(), foutWrite, jBuff.get(), 0, count);
+            }
+
+            env->CallVoidMethod(zipIn.get(), zinCloseEntry);
+            env->CallVoidMethod(fout.get(), foutClose);
+        }
+    }
+
+    JavaWrapper<jobject> fileToDelete(env, env->NewObject(fileClass, fileCtor, destination.get(), zipFile.get()));
+    if (env->CallBooleanMethod(fileToDelete.get(), fileExists)) {
+        jboolean success = (env->CallBooleanMethod(fileToDelete.get(), deleteFile));
+        if (!success) {
+            mbgl::Log::Warning(mbgl::Event::General, "Failed to delete file entry %s", zipFilePath.c_str());
+        }
+    }
+}
+
+} // namespace
+
+bool running = false;
+bool done = false;
+ALooper* looper = NULL;
+
+void runner() {
+    std::vector<std::string> arguments = {"mbgl-benchmark-runner",
+                                          "--benchmark_repetitions=3",
+                                          "--benchmark_format=json",
+                                          "--benchmark_out=/sdcard/benchmark/results/results.json"};
+    std::vector<char*> argv;
+    for (const auto& arg : arguments) {
+        argv.push_back(const_cast<char*>(arg.data()));
+    }
+    argv.push_back(nullptr);
+
+    mbgl::Log::Info(mbgl::Event::General, "Start BenchmarkRunner");
+    int status = mbgl::runBenchmark(argv.size(), argv.data());
+    mbgl::Log::Info(mbgl::Event::General, "BenchmarkRunner finished with status: '%d'", status);
```

Comment:
What does this status mean? should we use status result as input for `changeState(env, app, true);`

Reply:
The returned status is actually always 0, so we shouldn't use it as input for changeState().

Reply:
I see, then probably we do not need this variable.